Feature/gitapi

junbeomlee opened the pull request:
smartContract package setup

junbeomlee committed:
Merge branch 'feature/consensus' into develop

junbeomlee committed:
set package and add git_api
.go

hackurity01 commented:
아무래도 Clone은 git clone이랑 겹쳐서 너무 헷갈리는거 같습니다 ㅋㅋ

그리고 버전관리를 위해 clone 하고 version 관리 파일을 추가하면 어떨까 싶습니다
전에 그대로 클론하고 커밋하니까 커밋 해쉬가 (사용자 깃 커밋 해쉬랑) 똑같이 나왔던거 같은데 위 코드에서는 어떤지 모르겠네요

junbeomlee commented:
저도 버전관리 파일 고민하고있었는데 어떻게 하리 고민중... 똑같이 커밋 해쉬 나오는거는 같습니당 스마트컨트랙트 안에 그냥 파일을 하나 만들지 json같은걸로 아니면 leveldb에 메타정보를 저장할지? key.value기반으로

hackurity01 commented:
json으로 저장하면 좋겠네요 나중에 코드에서 불러와서 확인하기도 좋고
다양한 정보가 저장될 수 있을거 같은데
일단 생각나는건 시간, 이름, deployer 정도?

ghost commented:
앗.. Clone이 git clone하고 다른건가요??

junbeomlee commented:
git clone한후에 디비에 저장하는 로직이 추가되어있어서 함수명을 Clone 과 DB저장을 모두 포함하는 명칭으로 해서 더 명확하게 표현하자는 의미입니다!

hackurity01 commented:
Publish() 어떤가요????!!

junbeomlee commented:
생각해보니까 clone은 git_api에서 하고, 저장은 Deploy쪽에서 해주는게 맞을 듯? return으로 파일 path, id 등등을 리턴해주고 그걸로 Deploy에서 저장하는게 맞을 듯 하네여 생각해보니까 어떻게 생각하시나여? @hackurity01 @yojkim @owljoa

ghost commented:
@junbeomlee 스마트 컨트랙 생성해서 등록할 때,, 기존 저장소(?)를 clone 해와서 생성한 스마트 컨트랙을 Deploy(add, commit, push?) 하는 건가요??

junbeomlee commented:
기존저장소라기 보다는 Deploy하고 싶은 smartContract가 존재하는 저장소에서 clone한 스마트 컨트랙트를 tesseract를 이용해서 docker container에 올리는게 Deploy입니다.

ghost commented:
@junbeomlee 아하 그럼 스마트 컨트랙을 가상환경(docker container)에서 실행해보는 과정의 시작점이 Deploy인가요?

hackurity01 commented:
deploy를 하면
사용자의 깃에 있는 smartcontract를 clone해서 peer(로컬)에 저장하고 다시 공식 smartcontract 저장소(auth git)에 올립니다(push).
그리고 smartcontract를 나중에 실행할 수 있도록(query나 invoke) 해당 smartcontract를 가지는 docker container를 생성합니다(요 부분이 tesseract 역할).
이 과정이 deploy 과정입니다.
*더 구체적인건 제가 정리해서 공유드리겠습니다!

hackurity01 commented:
@junbeomlee
위에서 말한 대로면 git 관련 코드는 api가 아니고 domain으로 가야되는거 아닌가염?

ghost commented:
앗,, 기존 코드가 legacy에 있었군요..! 
@hackurity01 님 설명보고 deploy 코드보니 조금 이해됐습니다.
감사합니다 :)

junbeomlee commented:
@hackurity01 
외부 서비스 요청은 api레벨 or infra level 로 보는게 맞을듯!

junbeomlee commented:
@hackurity01  infra쪽인듯?..

junbeomlee commented:
https://github.com/VaughnVernon/IDDD_Samples 여기 example보면 협업 서비스는(외부 api call)은 infra레벨에서 구현해서 interface를 api레벨로 넣어줍니다!

junbeomlee commented:
@hackurity01  구현체는 infra가 맞는데 interface는 domain layer에 있어야 할 수도 있겠다.. 흠... 일단 내 생각은 구현체는 infra에 인터페이스는 api레벨로 보이는데 이 git_api가 domain logic을 수행하는지 안하는지가 가장 메인 포인트라고 하는데, 단순 git clone을 감싼거라 내 db에 저장하는 로직을 deploy로 옮기는게 맞는거 같아서 그렇게 보면 일다은 api레벨로 보인당.. 아직 좀더 공부해야봐야겠지만 일단 infra에구현체 두고 얘기해보면서 조정하는게 맞을듯!!

junbeomlee committed:
Clone function and add testcase

junbeomlee committed:
change sc to itcode and make itcodestore interface

junbeomlee commented:
@hackurity01 @owljoa 추가 푸쉬 했습니당!

hea9549 committed:
Fix : fix get last commit hash
smartcontract/infra/service/itrcode_store_git_api
공부하면서 소스코드랑 go doc 읽어봤는데
commitObjects로 commiter를 얻어오면 unsorted 라고 하네요
(https://godoc.org/gopkg.in/src-d/go-git.v4#Repository.CommitObjects)
reference를 얻어올때는 정렬되서 온다고 하니
HEAD 기준으로 last commit을 가져오는걸로 바꿔봤습니다~

junbeomlee committed:
Merge pull request #153 from hea9549/Feature/gitapi_haesung

Feature/gitapi, Fix : fix get last commit hash

junbeomlee committed:
delete init interface from itcode_store_api

junbeomlee committed:
add remoteChange clone createRepo

junbeomlee committed:
smartcontract to itcode

junbeomlee committed:
clone and push added

junbeomlee commented:
@hea9549 @hackurity01 체크체크!!

junbeomlee committed:
gitapi Push error fix

junbeomlee committed:
backup store interface added

junbeomlee committed:
refactor backup store git api

junbeomlee committed:
change itcode -> icodeMeta

junbeomlee commented:
@hackurity01 @owljoa @hea9549 다시한번 체크체크요!

hea9549 commented on smartcontract/infra/service/itcode_git_store_api.go:
```diff
@@ -76,6 +87,66 @@ func (g GitApi) Push(itCode itcode.ItCode) error {
+	return nil
+}
+
+func (g GitApi) ChangeRemote(path string, originUrl string) error {
```
@junbeomlee 
1. defaultBackUpGitUrl  이 AuthURL 말하는거맞져? 
auth랑 backup 에 혼란이오는데 authUrl이 맞으면 defaultAuthGitUrl로 바꾸는게 어때여 ?

2. ChangeRemote 이 단독으로 외부에서 호출될일이 있나요?
++ config 에 명시된 auth url 말고 다른 git server 로 분산될수있는건가요 ????private 이 맞지않나 해서 써봐욤

hea9549 replied:
@junbeomlee 오잉
